crates/modelardb_common/src/metadata/mod.rs

```diff
@@ -245,72 +1143,967 @@ mod tests {
+        initialize_model_table_with_tag_values(&mut metadata_manager, &["tag_value1"]).await;
+
+        // Lookup the univariate ids for the fallback column by only requesting tag columns.
+        let univariate_ids = metadata_manager
+            .compute_univariate_ids_using_fields_and_tags("model_table", Some(&vec![0]), 10, &[])
+            .await
+            .unwrap();
+
+        assert_eq!(1, univariate_ids.len());
+    }
+
+    #[tokio::test]
+    async fn test_compute_the_univariate_ids_for_a_specific_field_column_for_model_table() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let mut metadata_manager = try_new_sqlite_table_metadata_manager(temp_dir.path())
+            .await
+            .unwrap();
+
+        // Save a model table to the metadata database.
+        initialize_model_table_with_tag_values(
+            &mut metadata_manager,
+            &["tag_value1", "tag_value2"],
+        )
+        .await;
+
+        // Lookup all univariate ids for the table by not requesting ids for columns.
+        let univariate_ids = metadata_manager
+            .compute_univariate_ids_using_fields_and_tags("model_table", None, 10, &[])
+            .await
+            .unwrap();
+        assert_eq!(4, univariate_ids.len());
+
+        // Lookup univariate ids for the table by requesting ids for a specific field column.
+        let univariate_ids = metadata_manager
+            .compute_univariate_ids_using_fields_and_tags("model_table", Some(&vec![1]), 10, &[])
+            .await
+            .unwrap();
+        assert_eq!(2, univariate_ids.len());
+    }
+
+    #[tokio::test]
+    async fn test_compute_the_univariate_ids_for_a_specific_tag_value_for_model_table() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let mut metadata_manager = try_new_sqlite_table_metadata_manager(temp_dir.path())
+            .await
+            .unwrap();
+
+        // Save a model table to the metadata database.
+        initialize_model_table_with_tag_values(
+            &mut metadata_manager,
+            &["tag_value1", "tag_value2"],
+        )
+        .await;
+
+        // Lookup all univariate ids for the table by not requesting ids for a tag value.
+        let univariate_ids = metadata_manager
+            .compute_univariate_ids_using_fields_and_tags("model_table", None, 10, &[])
+            .await
+            .unwrap();
+        assert_eq!(4, univariate_ids.len());
+
+        // Lookup univariate ids for the table by requesting ids for a specific tag value.
+        let univariate_ids = metadata_manager
+            .compute_univariate_ids_using_fields_and_tags(
+                "model_table",
+                None,
+                10,
+                &[("tag", "tag_value1")],
+            )
+            .await
+            .unwrap();
+        assert_eq!(2, univariate_ids.len());
+    }
+
+    async fn initialize_model_table_with_tag_values(
+        metadata_manager: &mut TableMetadataManager<Sqlite>,
```

Comment:
Maybe renamed to `table_metadata_manager` for consistency.

Reply:
Fixed.